src/audio/pcm_converter/pcm_converter_generic.c

```diff
@@ -130,6 +130,37 @@ static void pcm_convert_s32_to_s24(struct comp_buffer *source,
 
 #endif /* CONFIG_FORMAT_S24LE && CONFIG_FORMAT_S32LE */
 
+#ifdef CONFIG_FORMAT_FLOAT && (CONFIG_FORMAT_S16LE || CONFIG_FORMAT_S24LE || CONFIG_FORMAT_S32LE)
+
+static inline int32_t _pcm_shift_left(int32_t d, int32_t a);
+static int32_t _pcm_convert_f_to_i(int32_t src);
+static int32_t _pcm_convert_i_to_f(int32_t src);
+
+static inline int32_t _pcm_shift_left(int32_t d, int32_t a)
+{
+	return (a >= 0) ? (d << a) : (d >> -a);
```

Comment:
This function could overflow with positive a. Could with this usage cause it exceed int32_t range?